backend/btrixcloud/migrations/migration_0020_org_storage_refs.py

```diff
@@ -0,0 +1,79 @@
+"""
+Migration 0020 - New Storage Ref System
+"""
+from btrixcloud.migrations import BaseMigration
+
+
+MIGRATION_VERSION = "0020"
+
+
+class Migration(BaseMigration):
+    """Migration class."""
+
+    def __init__(self, mdb, migration_version=MIGRATION_VERSION):
+        super().__init__(mdb, migration_version)
+
+    async def migrate_up(self):
+        """Perform migration up.
+
+        Convert storages to new storage ref system
+        - If default storage, convert to string
+        - If custom storage, add storage, convert to new name
+        """
+        # pylint: disable=duplicate-code, broad-exception-caught
+        mdb_orgs = self.mdb["organizations"]
+        async for org in mdb_orgs.find({"storage.custom": None}):
+            oid = org["_id"]
+            storage = org["storage"]
+
+            if storage.get("type") == "default":
+                update_dict = {
+                    "storage": {"name": storage.get("name"), "custom": False}
+                }
+
+            elif storage.get("type") == "s3":
+                update_dict = {
+                    "storage": {"name": storage.get("name"), "custom": True},
+                    "customStorages": {"custom": storage},
+                }
+
+            try:
+                await mdb_orgs.find_one_and_update({"_id": oid}, {"$set": update_dict})
+            except Exception as err:
+                print(f"Error updating storage for {oid}: {err}", flush=True)
+
+        # CrawlFile Migrations
+        mdb_crawls = self.mdb["crawls"]
+        async for crawl in mdb_crawls.find({"files.def_storage_name": {"$ne": None}}):
```

Comment:
I think we need to run this on all files (crawl and profile), even if `def_storage_name` is `None`. Otherwise, we get the following error when trying to list profiles as an example:

```
  File "/app/btrixcloud/profiles.py", line 229, in list_profiles
    profiles = [Profile.from_dict(res) for res in results]
  File "/app/btrixcloud/profiles.py", line 229, in <listcomp>
    profiles = [Profile.from_dict(res) for res in results]
  File "/app/btrixcloud/db.py", line 214, in from_dict
    return cls(**data)
  File "/usr/local/lib/python3.10/site-packages/pydantic/main.py", line 341, in __init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 1 validation error for Profile
resource -> storage
  field required (type=value_error.missing)
```

Reply:
Profile files at least made in current main have `def_storage_name` set to `None` but still need `storage` added in migration.